### ISSUE TITLE
[TypeScript] Bug: Token Authentication headers missing in typescript-angular and typescript-fetch client stubs

### ISSUE THREAD

**Issue by RS-Sautter** [swagger-api/swagger-codegen#9493]
##### Description

When I set the accessToken property in the configuration, the token does not get passed to the server with the request headers.

##### Swagger-codegen version

current version on editor.swagger.io

##### Related issues/PRs

#5656

**Comment by calcypher** [comment]
The same problem is present on PHP Client: security OpenAPI tags aren't "translated" into code, so there is no Authorization header on requests.
I've produced a Java CLient with the same OpenAPI and I've verified that this code managed Authorization headers.
So the problem is for PHP and TypeScript.

Is it possible to include it in this issue or it's necessary to open another issue for PHP?

**Comment by RS-Sautter** [comment]
Since it is a separate generator I would recommend opening another issue for it.

**Comment by louisl** [comment]
I've just run into this problem. What are my short term options for TypeScript with an Open API 3 spec?

This did used to work on with a swagger 2 spec.

**Comment by verbruggenalex** [comment]
Maybe it should be `isBearer` like in ASP?

https://github.com/swagger-api/swagger-codegen/blob/20d863d532998324d1a2d80d9d3e890b2d3184ed/modules/swagger-codegen/src/main/resources/aspnetcore/Startup.mustache#L64

**Comment by louisl** [comment]
Appears to be fixed  for typescript using 3.0.14, I can now drop the oAuthNoScopes workaround. Thanks whoever fixed it.

**Comment by lalli-flores** [comment]
I'm using generator version v3.0.22 for typescript-fetch, and I'm still seeing having this issue

**Comment by humbertowoody** [comment]
Hey @lalli-flores i'm running into the same issue, however I found a workaround:

Set your `bearer` configuration to:

```yaml
  securitySchemes:
    bearer:
      type: oauth2
      flows: 
        implicit:
          authorizationUrl: https://api.example.com/oauth2/authorize
          scopes: {}
```

Looking at https://github.com/swagger-api/swagger-codegen/blob/5a6d68716bc5a5f6fda10017ff380c382183d0c0/modules/swagger-codegen/src/main/resources/typescript-fetch/api.mustache#L138 there's clearly an error in which oAuth is the only way to get a working Bearer token. Hopefully this could be fixed, in the meantime you can use the above hack to override it! 

Best of luck! :)

**Comment by jannis6023** [comment]
Hi, may anyone be so nice to fix this issue? It's open since 12th june **2019**!!

**Comment by kit0kat** [comment]
Still not fixed in swagger-codegen-cli-3.0.36 for typescript-fetch.
As a workaround you can also use apiKey:

`"securitySchemes": {
      "Bearer Token": {
        "type": "apiKey",
        "name": "Authorization",
        "in": "header"
      }
    }`

I would also appreciate it if HTTP Bearer would work properly.

**Comment by maskaravivek** [comment]
Yes, it would be great to have a proper fix for it instead of relying on a workaround. Any timelines on when this issue would be fixed? :)

**Comment by oharkins** [comment]
Any Update to this?

**Comment by malloy045** [comment]
> Yes, it would be great to have a proper fix for it instead of relying on a workaround. Any timelines on when this issue would be fixed? :)

+1

The `apiKey` workaround doesn't work when your API expects a separate header named `apiKey` :)

I was able to get unblocked by modifying my `swagger.json` using the "OAuth2" workaround (`"type": "oauth2"`), but this isn't sustainable.

**Comment by pierre-mike-pge** [comment]
I think this is still an issue for me as well

**Comment by wisaac407** [comment]
Any updates on this? I'm still seeing this issue
